code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -0,0 +1,76 @@
+/obj/structure/fermenting_barrel
+	name = "wooden barrel"
+	desc = "A large wooden barrel. You can ferment fruits and such inside it, or just use it to hold liquid."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	density = TRUE
+	anchored = FALSE
+	container_type = DRAINABLE | AMOUNT_VISIBLE
+	var/open = FALSE
+	pressure_resistance = 2 * ONE_ATMOSPHERE
+	max_integrity = 300
+	var/speed_mod = 1 //How fast it distills. Defaults to 100% (1.0). Lower is better.
+
+/obj/structure/fermenting_barrel/Initialize()
+	create_reagents(300) //Bluespace beakers, but without the portability or efficiency in circuits.
+	. = ..()
+
+/obj/structure/fermenting_barrel/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>It is currently [open?"open, letting you pour liquids in.":"closed, letting you draw liquids from the tap."]</span>")
+
+/obj/structure/fermenting_barrel/proc/makeWine(obj/item/reagent_containers/food/snacks/grown/fruit)
+	if(fruit.reagents)
+		fruit.reagents.trans_to(src, fruit.reagents.total_volume)
+	var/amount = fruit.seed.potency / 4
+	if(fruit.distill_reagent)
+		reagents.add_reagent(fruit.distill_reagent, amount)
+		qdel(fruit)
+		playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
+		return
+	else
+		var/data = list()
+		data["names"] = list("[initial(fruit.name)]" = 1)
+		data["color"] = fruit.filling_color
+		data["boozepwr"] = fruit.wine_power
+		if(fruit.wine_flavor)
+			data["tastes"] = list(fruit.wine_flavor = 1)
+		else
+			data["tastes"] = list(fruit.tastes[1] = 1)
+		reagents.add_reagent("fruit_wine", amount, data)
+		qdel(fruit)
+		playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
+		return
+
+/obj/structure/fermenting_barrel/attackby(obj/item/I, mob/user, params)
+	var/obj/item/reagent_containers/food/snacks/grown/fruit = I
+	if(istype(fruit))
+		if(!fruit.can_distill)
+			to_chat(user, "<span class='warning'>You can't distill this into anything...</span>")
+			return TRUE
```

Comment:
what does return FALSE do? looks like you'd use it here since it fails.

Reply:
@vuonojenmustaturska told me to return TRUE so that it interrupted attack cycle.

Reply:
oh ok